plugins/default/lib/component_health.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+import sys
+import json
+import argparse
+import urllib2
+
+import nagios
+
+
+class RequestError(Exception):
+
+    def __init__(self, message):
+        self.message = message
+
+
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-H",
+        "--host",
+        action="store",
+        required=True,
+        help="Host")
+    parser.add_argument(
+        '-P',
+        '--port',
+        action="store",
+        required=True,
+        help="Port")
+    parser.add_argument(
+        "-E",
+        "--endpoint",
+        action="store",
+        required=False,
+        help="Endpoint",
+        default="/sys/info/health")
+    return parser
+
+
+def do_request(host, port, endpoint):
+    url = 'http://%s:%s%s' % (host, port, endpoint)
+    try:
+        data = urllib2.urlopen(url).read()
+    except urllib2.HTTPError as error:
+        if error.code == 500:
+            # For some reason the health endpoint returns a http status code of
+            # 500 when the health isn't ok, so we have to check the http error
+            # for a health response here
+            try:
+                data = error.read()
+                json.loads(data)
+            except ValueError:
+                raise RequestError(error)
+        else:
+            raise RequestError(error)
+    except urllib2.URLError:
+        raise RequestError(
+            'The health endpoint at "%s" is not contactable. This could indicate that the service is down or the url is invalid!' %
```

Comment:
Shouldn't be just the error, but a user friendly message + details.

Imagine the error is a connection timeout, or DNS error. It would be useful to be able to differentiate that from the health endpoint reporting a problem.

If we omit the original error, we can't determine what actually happened.